app/models/user.rb

```diff
@@ -0,0 +1,25 @@
+class User < ApplicationRecord
+  before_save :downcase_email
+
+  validates :name, presence: true, length: {maximum: Settings.name.length.max}
+
+  validates :email, presence: true,
+    length: {
+      minimum: Settings.email.length.min,
+      maximum: Settings.email.length.max
+    },
+    format: {with: Settings.email.valid_regex},
+    uniqueness: {case_sensitive: false}
+
+  validates :password, presence: true,
+    length: {minimum: Settings.password.length.min},
+    if: :password
+
+  has_secure_password
+
+  private
+
+  def downcase_email
+    @email.downcase!
```

Comment:
k cần @ nhé e
phân biệt downcase vs downcase!

Reply:
downcase và downcase! đều làm các ký tự chữ cái trong String trở thành chữ cái thường
ví dụ với string str
str.downcase sẽ trả về kết quả là str sau khi downcase toàn bộ chữ cái, nhưng giá trị của str thì không thay đổi
str.downcase! sẽ trả về như str.downcase nếu có sự thay đổi diễn ra, còn không thay đổi gì thì str.downcase! sẽ trả về nil. str.downcase! sẽ thay đổi giá trị của str như kết quả trả về nếu hàm được thực hiện.